--- src/Cedar/Cedar.h	2014-10-22 20:56:31.000000000 +0900
+++ src/Cedar/Cedar.h	2014-10-28 14:15:57.050268711 +0900
@@ -510,22 +510,25 @@
 #define	LOG_ENGINE_BUFFER_CACHE_SIZE_MAX	(10 * 1024 * 1024)	// Write cache size
 
 // Constant such as a file name
-#define	SERVER_LOG_DIR_NAME			"@server_log"
+#ifndef SOFTETHER_LOG_DIR
+#define SOFTETHER_LOG_DIR           "/var/log/softether/"
+#endif
+#define	SERVER_LOG_DIR_NAME			SOFTETHER_LOG_DIR "server_log"
 #define	BRIDGE_LOG_DIR_NAME			SERVER_LOG_DIR_NAME
 #define	SERVER_LOG_PERFIX			"vpn"
 
-#define	HUB_SECURITY_LOG_DIR_NAME	"@security_log"
-#define	HUB_SECURITY_LOG_FILE_NAME	"@security_log/%s"
+#define	HUB_SECURITY_LOG_DIR_NAME	SOFTETHER_LOG_DIR "security_log"
+#define	HUB_SECURITY_LOG_FILE_NAME	SOFTETHER_LOG_DIR "security_log/%s"
 #define	HUB_SECURITY_LOG_PREFIX		"sec"
-#define	HUB_PACKET_LOG_DIR_NAME		"@packet_log"
-#define	HUB_PACKET_LOG_FILE_NAME	"@packet_log/%s"
+#define	HUB_PACKET_LOG_DIR_NAME		SOFTETHER_LOG_DIR "packet_log"
+#define	HUB_PACKET_LOG_FILE_NAME	SOFTETHER_LOG_DIR "packet_log/%s"
 #define	HUB_PACKET_LOG_PREFIX		"pkt"
 
-#define	NAT_LOG_DIR_NAME			"@secure_nat_log"
-#define	NAT_LOG_FILE_NAME			"@secure_nat_log/%s"
+#define	NAT_LOG_DIR_NAME			SOFTETHER_LOG_DIR "secure_nat_log"
+#define	NAT_LOG_FILE_NAME			SOFTETHER_LOG_DIR "secure_nat_log/%s"
 #define	NAT_LOG_PREFIX				"snat"
 
-#define	CLIENT_LOG_DIR_NAME			"@client_log"
+#define	CLIENT_LOG_DIR_NAME			SOFTETHER_LOG_DIR "client_log"
 #define	CLIENT_LOG_PREFIX			"client"
 
 // Packet log settings
@@ -560,8 +563,8 @@
 #define	DISK_FREE_CHECK_INTERVAL_DEFAULT	(5 * 60 * 1000)
 
 // Simple log
-#define TINY_LOG_DIRNAME			"@tiny_log"
-#define TINY_LOG_FILENAME			"@tiny_log/%04u%02u%02u_%02u%02u%02u.log"
+#define TINY_LOG_DIRNAME			SOFTETHER_LOG_DIR "tiny_log"
+#define TINY_LOG_FILENAME			SOFTETHER_LOG_DIR "tiny_log/%04u%02u%02u_%02u%02u%02u.log"
 
 
 //////////////////////////////////////////////////////////////////////
@@ -574,7 +577,7 @@
 //#define CE_SNAPSHOT_INTERVAL		((UINT64)(3000))
 #define CE_SNAPSHOT_POLLING_INTERVAL	(1 * 1000)
 #define CE_SNAPSHOT_POLLING_INTERVAL_LICENSE	(30 * 1000)
-#define CE_SNAPSHOT_DIR_NAME		"@carrier_log"
+#define CE_SNAPSHOT_DIR_NAME		SOFTETHER_LOG_DIR "carrier_log"
 #define CE_SNAPSHOT_PREFIX			"carrier"
 
 
@@ -584,6 +587,8 @@
 // 
 //////////////////////////////////////////////////////////////////////
 
+#define SOFTETHER_SETTING_DIR       "/etc/softher/"
+
 // Administrator Username
 #define	ADMINISTRATOR_USERNAME		"administrator"
 // Maximum value of random size
@@ -591,7 +596,7 @@
 // Expiration date of random size cache
 #define	RAND_SIZE_CACHE_EXPIRE		(24 * 60 * 60 * 1000)
 // Management allowed IP address list file name
-#define	ADMINIP_TXT					"@adminip.txt"
+#define	ADMINIP_TXT					SOFTETHER_SETTING_DIR "adminip.txt"
 
 #define NON_SSL_MIN_COUNT			60
 #define NON_SSL_ENTRY_EXPIRES		(10 * 60 * 1000)
