--- v4.19-9599/src/Mayaqua/Unix.c	2015-10-19 20:47:50.000000000 +0900
+++ v4.19-9599/src/Mayaqua/Unix.c	2015-11-30 21:40:19.995716149 +0900
@@ -2222,7 +2222,8 @@
 #endif	// RLIMIT_NPROC
 
 	// Write a value to the threads-max of the proc file system
-	o = UnixFileCreate("/proc/sys/kernel/threads-max");
+	//cause violation during build on gentoo, so disable this by the environment variable
+	o = getenv("DISABLE_LINUX_THREADS_MAX_SETTING") ? NULL : UnixFileCreate("/proc/sys/kernel/threads-max");
 	if (o != NULL)
 	{
 		char tmp[128];
